pkg/daemon/ceph/osd/volume.go

```diff
@@ -108,6 +106,9 @@ func (a *OsdAgent) initializeDevices(context *clusterd.Context, devices *DeviceO
 		if device.Data == -1 {
 			logger.Infof("configuring new device %s", name)
 			deviceArg := path.Join("/dev", name)
+			if device.Config.OSDsPerDevice > 1 {
+				osdsPerDeviceCount = sanitizeOSDsPerDevice(device.Config.OSDsPerDevice)
```

Comment:
This changes the overall default instead of only applying to this OSD. Seems like you need to keep a different variable for the count specific to this device.

Reply:
you are right.  good catch.  I will change this to use a variable local to each device iteration